js/.eslintrc.js

```diff
@@ -0,0 +1,233 @@
+/* eslint-disable @typescript-eslint/indent */
+module.exports = {
+    env: {
+        browser: true,
+        es6: true,
+    },
+    extends: [
+        "plugin:@typescript-eslint/recommended",
+        "plugin:@typescript-eslint/recommended-requiring-type-checking",
+    ],
+    parser: "@typescript-eslint/parser",
+    parserOptions: {
+        project: "tsconfig.json",
+        createDefaultProgram: true,
+        sourceType: "module",
+    },
+    plugins: [
+        "@typescript-eslint",
+        "@typescript-eslint/tslint",
+    ],
+    rules: {
+        "@typescript-eslint/array-type": [
+            "error",
+            {
+                default: "array-simple",
+            },
+        ],
+        "@typescript-eslint/ban-types": [
+            "error",
+            {
+                types: {
+                    Object: {
+                        message: "Avoid using the `Object` type. Did you mean `object`?",
+                    },
+                    Function: {
+                        message: "Avoid using the `Function` type. Prefer a specific function type, like `() => void`.",
+                    },
+                    Boolean: {
+                        message: "Avoid using the `Boolean` type. Did you mean `boolean`?",
+                    },
+                    Number: {
+                        message: "Avoid using the `Number` type. Did you mean `number`?",
+                    },
+                    String: {
+                        message: "Avoid using the `String` type. Did you mean `string`?",
+                    },
+                    Symbol: {
+                        message: "Avoid using the `Symbol` type. Did you mean `symbol`?",
+                    },
+                },
+            },
+        ],
+        "@typescript-eslint/camelcase": "off",
+        "@typescript-eslint/consistent-type-definitions": "error",
+        "@typescript-eslint/dot-notation": "error",
+        "@typescript-eslint/explicit-member-accessibility": [
+            "error",
+            {
+                accessibility: "explicit",
+            },
+        ],
```

Comment:
```suggestion
        "@typescript-eslint/explicit-member-accessibility": "no-public",
```

redundant `public` keywords are redundant